logparser/logparser.py

```diff
@@ -156,11 +156,15 @@ def _parse_log(self):
                                               True)
 
         # While there is a new line, parse it.
-        line = True  # For the first condition.
-        while line:
+        line = ""
+        while line is not None:
             # If the line contains non-UTF8 chars it could raise an exception.
             self.state['input_line'] += 1
-            line = device.read_line().rstrip("\r\n")
+            line = device.read_line()
+
+            # Remove end of lines
+            if line:
```

Comment:
This is because we want to remove new line chars only when `line` is not `None` and we don't want to set `line` to an empty string to distinguish between empty line and EOF.